src/ORM/ManyManyThroughQueryManipulator.php

```diff
@@ -36,19 +36,37 @@ class ManyManyThroughQueryManipulator implements DataQueryManipulator
      */
     protected $foreignKey;
 
+    /**
+     * Foreign class 'from' property. Normally not needed unless polymorphic.
+     *
+     * @var string
+     */
+    protected $foreignClass;
+
+    /**
+     * Class name of instance that owns this list
+     *
+     * @var string
+     */
+    protected $parentClass;
+
     /**
      * Build query manipulator for a given join table. Additional parameters (foreign key, etc)
      * will be infered at evaluation from query parameters set via the ManyManyThroughList
      *
      * @param string $joinClass Class name of the joined dataobject record
      * @param string $localKey The key in the join table that maps to the dataClass' PK.
      * @param string $foreignKey The key in the join table that maps to joined class' PK.
+     * @param string $foreignClass the 'from' class name
+     * @param string $parentClass Name of parent class. Subclass of $foreignClass
      */
-    public function __construct($joinClass, $localKey, $foreignKey)
+    public function __construct($joinClass, $localKey, $foreignKey, $foreignClass, $parentClass)
```

Comment:
Can we make these new arguments optional? This is a fairly internal API, but I don’t think it’s totally unreasonable to expect that someone could be using this class directly

Reply:
They aren't optional though. ;) This class needs it to work.

Reply:
I’m just being a militant semversionist. `... $foreignClass = null, $parentClass = null)` should work and be backwards compatible shouldn’t it? I can’t spot any code in this class that would break if those props are just `null`

Reply:
null means unknown, so without those variables it may mistakenly not mock a polymorphic relation which is polymorphic. The ambiguity introduces a functional error, not a syntactic one.

Reply:
In this case I could just make it a warning, so sure.